extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -1195,20 +1194,31 @@ static boolean processHints(TemplateAnalysis templateAnalysis, String helperHint
         } else if (helperHint.startsWith(WhenSectionHelper.Factory.HINT_PREFIX)) {
             // If a value expression resolves to an enum we attempt to use the enum type to validate the enum constant  
             // This basically transforms the type info "ON<when:12345>" into something like "|org.acme.Status|.ON"
-            Integer valueExprId = Integer
-                    .valueOf(helperHint.substring(WhenSectionHelper.Factory.HINT_PREFIX.length(), helperHint.length() - 1));
-            Expression valueExpr = templateAnalysis.findExpression(valueExprId);
+            Expression valueExpr = findExpression(helperHint, WhenSectionHelper.Factory.HINT_PREFIX, templateAnalysis);
             if (valueExpr != null) {
                 Match valueExprMatch = generatedIdsToMatches.get(valueExpr.getGeneratedId());
                 if (valueExprMatch != null && valueExprMatch.clazz.isEnum()) {
                     match.setValues(valueExprMatch.clazz, valueExprMatch.type);
                     return true;
                 }
             }
+        } else if (helperHint.startsWith(SetSectionHelper.Factory.HINT_PREFIX)) {
+            Expression valueExpr = findExpression(helperHint, SetSectionHelper.Factory.HINT_PREFIX, templateAnalysis);
+            if (valueExpr != null) {
+                Match valueExprMatch = generatedIdsToMatches.get(valueExpr.getGeneratedId());
+                if (valueExprMatch != null) {
+                    match.setValues(valueExprMatch.clazz, valueExprMatch.type);
+                }
+            }
         }
         return false;
     }
 
+    private static Expression findExpression(String helperHint, String hintPrefix, TemplateAnalysis templateAnalysis) {
+        Integer id = Integer.valueOf(helperHint.substring(hintPrefix.length(), helperHint.length() - 1));
```

Comment:
Change it to `Integer.parseInt` and remove the unnecessary Integer autoboxing in `io.quarkus.qute.deployment.TemplatesAnalysisBuildItem.TemplateAnalysis#findExpression`